extensions/crossmodel-lang/src/language-server/cross-model-validator.ts

```diff
@@ -87,70 +88,78 @@ export class CrossModelValidator {
       const basenameWithoutExt = basename.slice(0, -extname.length);
-      if (basenameWithoutExt.toLowerCase() !== node.id.toLocaleLowerCase()) {
-         accept('warning', `Filename should match element id: ${node.id}`, {
-            node,
+      if (basenameWithoutExt.toLowerCase() !== identifiedObject.id?.toLocaleLowerCase()) {
+         accept('warning', `Filename should match element id: ${identifiedObject.id}`, {
+            node: identifiedObject,
             property: ID_PROPERTY,
             data: { code: CrossModelIssueCodes.FilenameNotMatching }
          });
       }
    }
 
-   protected checkUniqueGlobalId(node: AstNode, accept: ValidationAcceptor): void {
-      if (!this.isExportedGlobally(node)) {
+   // Check the uniqueness of ids of semantic root elements.
+   protected checkUniqueGlobalId(identifiedObject: IdentifiedObject, accept: ValidationAcceptor): void {
+      if (!isSemanticRoot(identifiedObject)) {
          return;
       }
-      const globalId = this.services.references.IdProvider.getGlobalId(node);
+      const globalId = this.services.references.IdProvider.getGlobalId(identifiedObject);
       if (!globalId) {
-         accept('error', 'Missing required id field', { node, property: ID_PROPERTY });
+         accept('error', 'Missing required id field', { node: identifiedObject, property: ID_PROPERTY });
          return;
       }
       const allElements = Array.from(this.services.shared.workspace.IndexManager.allElements());
       const duplicates = allElements.filter(description => description.name === globalId);
       if (duplicates.length > 1) {
-         accept('error', 'Must provide a unique id.', { node, property: ID_PROPERTY });
+         accept('error', 'Must provide a unique id.', { node: identifiedObject, property: ID_PROPERTY });
       }
    }
 
-   protected isExportedGlobally(node: AstNode): boolean {
-      // we export anything with an id from entities and relationships and all root nodes, see CrossModelScopeComputation
-      return isLogicalEntity(node) || isLogicalAttribute(node) || isRelationship(node) || isSystemDiagram(node) || isMapping(node);
-   }
-
-   protected checkUniqueNodeId(node: AstNode, accept: ValidationAcceptor): void {
+   // Check the uniqueness of ids of non-semantic root identified objects.
+   protected checkUniqueLocalId(node: AstNode, accept: ValidationAcceptor): void {
+      if (isLogicalEntity(node)) {
+         this.markDuplicateIds(node.attributes, accept);
+         this.markDuplicateIds(node.identifiers, accept);
+      }
       if (isSystemDiagram(node)) {
          this.markDuplicateIds(node.edges, accept);
          this.markDuplicateIds(node.nodes, accept);
       }
       if (isMapping(node)) {
          this.markDuplicateIds(node.sources, accept);
       }
+      if (isWithCustomProperties(node)) {
+         this.markDuplicateIds(node.customProperties, accept);
```

Comment:
This does not seem to work for custom properties within the target mappings of a mapping diagram.